From 864115eef3e5d230015043f72998c8c201efb36f Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 10 Sep 2018 09:58:43 +0800
Subject: [PATCH 4562/5242] MLK-19511-2 ARM64: dts: freescale: imx8qm: move
 cm4_intmux early_power_on to board dtb

commit  0e80fec3b09f2f6c30a8fea02fc9a9891061913d from
https://source.codeaurora.org/external/imx/linux-imx.git

When early_power_on is present in power domain dtb node, it
will be powered on during resume regardless of whether the
related module is enabled or NOT, this will cause cm4_intmux
always power ON after first time resume when cm4_intmux is
NOT enabled.

So move this early_power_on property to board level dtb, ONLY
when cm4_intmux is enabled, then this property is added.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi      |    2 --
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi  |    8 ++++++++
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index 00177b8..a9e4ebc 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -1363,7 +1363,6 @@
 				reg = <SC_R_M4_0_INTMUX>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_cm40>;
-				early_power_on;
 			};
 		};
 
@@ -1384,7 +1383,6 @@
 				reg = <SC_R_M4_1_INTMUX>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_cm41>;
-				early_power_on;
 			};
 		};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
index 5f4249b..adf09c1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
@@ -1081,6 +1081,10 @@
 	status = "okay";
 };
 
+&pd_cm40_intmux {
+	early_power_on;
+};
+
 &intmux_cm40 {
 	status = "okay";
 };
@@ -1094,6 +1098,10 @@
 	status = "okay";
 };
 
+&pd_cm41_intmux {
+	early_power_on;
+};
+
 &intmux_cm41 {
 	status = "okay";
 };
-- 
1.7.9.5

